### ISSUE TITLE
Completion of error handling

### ISSUE THREAD

**Issue by elfring** [hashcat/hashcat#3570]
Would you like to add more error handling for return values from functions like the following?
* [fclose](https://pubs.opengroup.org/onlinepubs/9699919799/functions/fclose.html "Close a stream") ⇒ [hc_fclose](https://github.com/hashcat/hashcat/blob/31758f09a8666fecf35dc93e136f516367e7ca9d/src/filehandling.c#L942)
* [hc_fprintf](https://github.com/hashcat/hashcat/blob/31758f09a8666fecf35dc93e136f516367e7ca9d/src/filehandling.c#L843 "Print formatted output") ⇒ [write_restore](https://github.com/hashcat/hashcat/blob/ea29cb5805f15e9d6b440a501a2f821bda4a487e/src/restore.c#L217)

**Comment by jsteube** [comment]
I was thinking we already catch errors whenever it was required. Sometimes we intentionally do not. Do you have any examples where it would be useful to do a change?

**Comment by elfring** [comment]
:thought_balloon: I suggest to avoid [ignorance of return values](https://cwe.mitre.org/data/definitions/252.html "Checking more return values") a bit more.
Would you like to detect every error situation as early as possible?

**Comment by jsteube** [comment]
Yeah typically as early as possible but in some cases not at all. It really depends on the example, that's why I was asking for that.

**Comment by elfring** [comment]
:thought_balloon: It is just safer to [handle also “exceptional” situations](https://wiki.sei.cmu.edu/confluence/display/c/EXP12-C.+Do+not+ignore+values+returned+by+functions "Do not ignore values returned by functions"), isn't it?

:crystal_ball: How will chances evolve to benefit any more also from [the means of aspect-oriented software development](https://en.wikipedia.org/wiki/Aspect-oriented_programming "Documentation")?

**Comment by elfring** [comment]
:thinking: How much would you like to care for file output failures as I indicated risks by my bug report?
